src/Razor/src/Microsoft.AspNetCore.Razor.LanguageServer/Formatting/CSharpOnTypeFormattingPass.cs

```diff
@@ -51,7 +54,33 @@ public async override Task<FormattingResult> ExecuteAsync(FormattingContext cont
             // Normalize and re-map the C# edits.
             var codeDocument = context.CodeDocument;
             var csharpText = codeDocument.GetCSharpSourceText();
-            var normalizedEdits = NormalizeTextEdits(csharpText, result.Edits, out var originalTextWithChanges);
+
+            var textEdits = result.Edits;
+            if (textEdits.Length == 0)
```

Comment:
This is due to a quirk which I wasn't previously aware of (or at least, it wasn't as obvious)

Before this change our on type formatting worked by calling Roslyn to do on type formatting, and then passing those edits into our formatting engine, to run through all of the formatting passes.

After this change, the call to Roslyn is in the `CSharpOnTypeFormattingPass`, because that is where logically it should belong (the call to Roslyn for whole document formatting is in `CSharpFormattingPass` and the call to WebTools for Html formatting is in `HtmlFormattingPass`). Unfortunately there are other things (snippts, completion etc.) which construct a set of edits and then call our formatting engine pretending to be on type formatting. 

This check means exactly what you've said, and is here to not break those scenarios. Obviously this is crazy though, and should be separated out, I just haven't worked out exactly how yet.

Reply:
Sounds like a big juicy comment.

Reply:
Probably more appropriate as an issue IMO, but I'm not really sure what I'm doing with this PR yet, so it might end up getting fixed in here. We'll see.

Reply:
Logged https://github.com/dotnet/razor-tooling/issues/6120